### DISCUSSION TITLE
Width of the progress bar not changing

### DISCUSSION THREAD

**Discussion by felixvanneste** [armandsalle/butter-slider#19] [Q&A]
Hello @armandsalle 

I have followed the different steps in the video you made with Alex Tourgis (https://www.youtube.com/watch?v=Txi6PbzENOg&list=PLXEka-n2gU9IN0g4W1MLrIqxuBNtNKYHI&index=7&t=723s&ab_channel=AlexTourgis).

The slider is working, but I can't get the progress bar to work.

If you want to have a look at the read only webflow project (bootom of the page) : https://preview.webflow.com/preview/allfc001-all-for-climate-new-website?utm_medium=preview_link&utm_source=designer&utm_content=allfc001-all-for-climate-new-website&preview=240f048af32f996eb3cff5a5051c4518&mode=preview

Worth mentioning : I am “just” a designer, so JS and others are obscure to me…

Thank you very much for your help !

**Comment by armandsalle** [comment]
Hello @felixvanneste !

Thank you for your feedback, can you share me a preview link ? It's easier for me to test it directly live ;)

Really nice design by the way 🔥 thank you for using Butter Slider

**Reply by felixvanneste** [reply]
Hello @armandsalle 

Thank you for your reply !

Here is the link : https://allfc001-all-for-climate-new-website.webflow.io/

**Reply by armandsalle** [reply]
Thank you, I found the problem. It's a bug with Butter Slider, in order for the progress bar to work it is necessary that the slidable div has an overflow-x: visible and the container div or the parent of the slidable div has an overflow: hidden.

It's with those CSS values that I can calculate the progression, but it's not convenient. You will need to refactor your HTML to get the effect you want.

**Reply by armandsalle** [reply]
You can find an example here : https://codesandbox.io/s/butter-slider-demo-rwxwi?file=/index.html

**Reply by felixvanneste** [reply]
Thanks for the help @armandsalle 

I managed to make it work… but it is not really in line with the rest of the website, as I can't use a fixed container with overflow visible. Not sure I will keep the progress bar.

I realised also that the container HAS to be on “overflow: visible" on mobile to be able to use the slider. So it means that the progress bar can't be used on mobile (and I think that Alex had the same issue, because the progress bar is not visible on mobile also on https://www.odisea-expedition.com/).

But I can't complain, it is already looking nice ! Thanks again.

PS : anyway you think V 1.2 can fix this ??

**Reply by felixvanneste** [reply]
Oh and sorry, one more thing : my slider doesn't work with the trackpad, and I can see that on https://www.odisea-expedition.com/ it is. Any idea ??

Merci !

**Reply by armandsalle** [reply]
😥😥 I really need to refactor my code and write a better doc! I’m looking for help so if you know good JS developers which are using Butter Slider, feel free to send them my email armand.salle@gmail.com 

if you want me to try to add BS directly on your project it will be a pleasure, just drop me a line. I hope we will find a solution for that work 👍

thank you for all your feedbacks, the road map for the next version is getting bigger and bigger. Not so simple as I was thinking 😭 I don’t know when it will be published but if you want the latest news, follow me on Twitter 🎉

**Reply by felixvanneste** [reply]
Hello @armandsalle No problem, I am already happy with what it can do now !
I will ask around if somebody can give a hand.

Thanks a lot !
